plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/tool/display/ToolManager.java

```diff
@@ -1003,23 +1014,38 @@ public void run() {
                             activeMap.forwardHistory();
                     }
                 };
-                forwardAction.setImageDescriptor(sharedImages
-                        .getImageDescriptor(ISharedImages.IMG_TOOL_FORWARD));
+                forwardAction.setImageDescriptor(
+                        getSharedImages().getImageDescriptor(ISharedImages.IMG_TOOL_FORWARD));
                 forwardAction.setText(Messages.ToolManager_forward);
                 forwardAction.setToolTipText(Messages.ToolManager_forward_tooltip);
                 forwardAction.setActionDefinitionId("org.eclipse.ui.navigate.forward"); //$NON-NLS-1$
             }
-            if (activeMap != ApplicationGIS.NO_MAP)
-                forwardAction.setEnabled(activeMap.getCommandStack().canRedo());
-            else
-                forwardAction.setEnabled(false);
+            setNavActionEnabledState(activeMap, forwardAction, true);
             return forwardAction;
         }finally{
             forwardLock.unlock();
         }
 
     }
 
+    private void setActionEnabledState(Map activeMap, IAction action, boolean isForwardAction) {
```

Comment:
isForwardAction is kind of confusing since its "undo/redo". Maybe "isRedo" in this case for the third method parameter?

Reply:
done (parameter name is now isRedoAction)